utils/awsFunctions.ts

```diff
@@ -0,0 +1,173 @@
+import {
+    IdentitystoreClient,
+    ListUsersCommand,
+    CreateUserCommand,
+    CreateGroupMembershipCommand,
+    ListUsersCommandInput,
+    CreateUserCommandInput,
+    CreateGroupMembershipCommandInput,
+  } from "@aws-sdk/client-identitystore";
+  import config from "config";
+  
+  // Define the configuration variables with proper types
+  const accessKeyId: string = config.get<string>("aws.access_key");
+  const secretAccessKey: string = config.get<string>("aws.secret_key");
+  const region: string = config.get<string>("aws.region");
+  const identityStoreId: string = config.get<string>("aws.identity_store_id");
+  
+  let client: IdentitystoreClient;
+  
+  /**
+   * Configures AWS SDK credentials and returns a singleton IdentitystoreClient instance.
+   * @returns {IdentitystoreClient} Singleton AWS Identitystore client
+   */
+  function configureAWSCredentials(): IdentitystoreClient {
+    if (!client) {
+      if (accessKeyId && secretAccessKey) {
+        client = new IdentitystoreClient({
+          region: region,
+          credentials: {
+            accessKeyId: accessKeyId,
+            secretAccessKey: secretAccessKey,
+          },
+        });
+      }
+    }
+    return client!;
+  }
+  
+  /**
+   * Lists users from AWS Identity Store filtered by username.
+   * @param {string} username - Username to filter users by
+   * @returns {Promise<string | null>} - UserId if found, otherwise null
+   */
+  export const fetchUserIdFromUsername = async (username: string): Promise<string | null> => {
+    const client = configureAWSCredentials();
+  
+    const getUserByIdCommand: ListUsersCommandInput = {
+      IdentityStoreId: identityStoreId,
+      Filters: [
+        {
+          AttributePath: "Username",
+          AttributeValue: username,
+        },
+      ],
+    };
+  
+    try {
+      const response = await client.send(new ListUsersCommand(getUserByIdCommand));
+  
+      if (response.Users && response.Users.length > 0) {
+        const userId = response.Users[0].UserId!;
+        return userId;
+      } else {
+        console.error("No users found for the provided username.");
+        return null;
+      }
+    } catch (err) {
+      throw new Error(`Failed to list users: ${err instanceof Error ? err.message : String(err)}`);
+    }
```

Comment:
Do we need the try-catch for the entire block of code or just the `client.send`?

Reply:
will update this

Reply:
Updated this